Release build with static runtime fails in Visual Studio 16.3 (2019)

Due to some peculiarities in the Botan build system, building a release version with static runtime fails with the latest Visual Studio version when using the following configure options:

    INFO: configure.py invoked with options "--cpu=x86_64 --with-debug-info --with-build-dir=C:\.conan\44fd0c\1 --msvc-runtime MT --disable-shared"

The Botan build system seems to add some of the same parameters to the linker as it does to the compiler. However, the MSVC linker does not support the options ``/bigobj`` and ``/MT``. In earlier linker versions, those options were ignored but unfortunately, the latest version fails in this case (``/MTd``, ``/MD``, and ``/MDd`` are still ignored by the linker). This is the current linker call and the resulting error:

    link /DEBUG /MT /bigobj C:\.conan\44fd0c\1\build\obj\cli\argon2.obj C:\.conan\44fd0c\1\build\obj\cli\asn1.obj C:\.conan\44fd0c\1\build\obj\cli\bcrypt.obj C:\.conan\44fd0c\1\build\obj\cli\cc_enc.obj C:\.conan\44fd0c\1\build\obj\cli\cli.obj C:\.conan\44fd0c\1\build\obj\cli\cli_rng.obj C:\.conan\44fd0c\1\build\obj\cli\codec.obj C:\.conan\44fd0c\1\build\obj\cli\compress.obj C:\.conan\44fd0c\1\build\obj\cli\encryption.obj C:\.conan\44fd0c\1\build\obj\cli\entropy.obj C:\.conan\44fd0c\1\build\obj\cli\hash.obj C:\.conan\44fd0c\1\build\obj\cli\hmac.obj C:\.conan\44fd0c\1\build\obj\cli\main.obj C:\.conan\44fd0c\1\build\obj\cli\math.obj C:\.conan\44fd0c\1\build\obj\cli\pbkdf.obj C:\.conan\44fd0c\1\build\obj\cli\pk_crypt.obj C:\.conan\44fd0c\1\build\obj\cli\psk.obj C:\.conan\44fd0c\1\build\obj\cli\pubkey.obj C:\.conan\44fd0c\1\build\obj\cli\sandbox.obj C:\.conan\44fd0c\1\build\obj\cli\speed.obj C:\.conan\44fd0c\1\build\obj\cli\timing_tests.obj C:\.conan\44fd0c\1\build\obj\cli\tls_client.obj C:\.conan\44fd0c\1\build\obj\cli\tls_http_server.obj C:\.conan\44fd0c\1\build\obj\cli\tls_proxy.obj C:\.conan\44fd0c\1\build\obj\cli\tls_server.obj C:\.conan\44fd0c\1\build\obj\cli\tls_utils.obj C:\.conan\44fd0c\1\build\obj\cli\tss.obj C:\.conan\44fd0c\1\build\obj\cli\utils.obj C:\.conan\44fd0c\1\build\obj\cli\x509.obj C:\.conan\44fd0c\1\botan.lib crypt32.lib user32.lib ws2_32.lib  /OUT:C:\.conan\44fd0c\1\botan-cli.exe
    Microsoft (R) Incremental Linker Version 14.23.28106.4
    Copyright (C) Microsoft Corporation.  All rights reserved.

    LINK : fatal error LNK1146: no argument specified with option '/MT'
    NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.23.28105\bin\HostX64\x64\link.EXE"' : return code '0x47a'
    Stop.

I think, the most reliable solution would be to modify the build system so that these usually ignored parameters are not added to the linker call (``/bigobj``, ``/MT``, ``/MTd``, ``/MD``, ``/MDd``).

--- a/configure.py
+++ b/configure.py
@@ -2147,6 +2147,12 @@ def create_template_vars(source_paths, build_paths, options, modules, cc, arch, 
         'mod_list': sorted([m.basename for m in modules])
     }
 
+    if cc.basename == 'msvc' and variables['cxx_abi_flags'] != '':
+        # MSVC linker doesn't support/need the ABI options,
+        # just transfer them over to just the compiler invocations
+        variables['cc_compile_flags'] = '%s %s' % (variables['cxx_abi_flags'], variables['cc_compile_flags'])
+        variables['cxx_abi_flags'] = ''
+
     variables['lib_flags'] = cc.gen_lib_flags(options, variables)
     variables['cmake_lib_flags'] = cmake_escape(variables['lib_flags'])
 

